libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -1941,6 +1941,15 @@ void IRGeneratorForStatements::endVisit(MemberAccess const& _memberAccess)
 			ContractDefinition const& contract = contractType.contractDefinition();
 			define(_memberAccess) << formatNumber(u256{contract.interfaceId()} << (256 - 32)) << "\n";
 		}
+		else if (member == "typehash")
+		{
+			Type const* arg = dynamic_cast<MagicType const&>(*_memberAccess.expression().annotation().type).typeArgument();
+			solAssert(!!arg);
+			StructType const* structType = dynamic_cast<StructType const*>(arg);
+			solAssert(!!structType);
```

Comment:
We can cast it straight away to `StructType`; the intermediate `arg` is not necessary.

Reply:
There is a chain:
1. `dynamic_cast<MagicType const&>`
2. `x.typeArgument()`
3. `dynamic_cast<StructType const*>`